src/Common/src/System/Net/Security/Unix/SafeFreeSslCredentials.cs

```diff
@@ -79,8 +85,22 @@ public SafeFreeSslCredentials(X509Certificate certificate, SslProtocols protocol
                     throw new NotSupportedException(SR.net_ssl_io_no_server_cert);
                 }
 
-                _certHandle = Interop.Crypto.X509Duplicate(cert.Handle);
+                _certHandle = Interop.Crypto.X509Duplicate(cert.Handle);                
                 Interop.Crypto.CheckValidOpenSslHandle(_certHandle);
+
+                using (var chain = new X509Chain())
+                {
+                    if (chain.Build(cert))
```

Comment:
The chain policy should be set to not care about almost anything; you just want the certs.  In fact, if TLSCertificateExtensions.BuildNewChain is in scope, you should just call that (with includeApplicationPolicy: false).